src/GraphQLParser/Lexer.cs

```diff
@@ -1,15 +1,18 @@
-﻿namespace GraphQLParser
+using System;
+
+namespace GraphQLParser
 {
-    public class Lexer : ILexer
+    public struct Lexer
     {
-        public ILexemeCache? Cache { get; set; }
+        public int? BufferSize { get; set; }
```

Comment:
By the way, the lexer can still allocate memory in the case of parsing strings with escape characters.